src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -599,6 +636,20 @@ private BoundExpression CreateAndPopulateArray(BoundCollectionExpression node, A
                             _factory.Binary(BinaryOperatorKind.Addition, indexTemp.Type, indexTemp, _factory.Literal(1)),
                             isRef: false,
                             indexTemp.Type));
+                },
+                (sideEffects, arrayTemp, spreadElement, rewrittenSpreadOperand) =>
```

Comment:
Consider naming these delegate parameters, especially now that there are multiple of them.

Reply:
Also, please spell out the types of these parameters. Same goes for the other similar delegates below.

Reply:
I wasn't sure what you meant by name the delegate parameters. Were you suggesting that I declare a named delegate type which has these parameter names?

Reply:
No, I mean specify the parameter name for each of these delegate arguments. What is the parameter this delegate is being passed to?